onprc_ehr/src/org/labkey/onprc_ehr/dataentry/IPC_SlidePrintingFormSection.java

```diff
@@ -0,0 +1,46 @@
+package org.labkey.onprc_ehr.dataentry;
+
+import org.labkey.api.ehr.EHRService;
+import org.labkey.api.ehr.dataentry.SimpleGridPanel;
+import org.labkey.api.view.template.ClientDependency;
+import java.util.List;
+
+/**
+ * User: Kolli
+ * Date: 4/01/2020
+ * Time: 10:36 AM
+ */
+public class IPC_SlidePrintingFormSection extends SimpleGridPanel
+{
+    public IPC_SlidePrintingFormSection()
+    {
+        this(EHRService.FORM_SECTION_LOCATION.Body);
+    }
+
+    public IPC_SlidePrintingFormSection(EHRService.FORM_SECTION_LOCATION location)
+    {
+        super("study", "IPC_SlidePrinting", "Slide Printing (No Sectioning)", location);
+        addClientDependency(ClientDependency.supplierFromPath("onprc_ehr/model/sources/IPC_SlidePrinting.js"));
+        addConfigSource("SlidePrinting");
+        addClientDependency(ClientDependency.supplierFromPath("onprc_ehr/model/sources/IPC_Child.js"));
+        addConfigSource("IPC_Child");
+
+        addClientDependency(ClientDependency.supplierFromPath("onprc_ehr/window/IPC_CopyFromSectionWindow.js"));
+    }
+
+    @Override
+    public List<String> getTbarButtons()
+    {
+        List<String> defaultButtons = super.getTbarButtons();
+
+        if (defaultButtons.contains("COPYFROMSECTION"))
+        {
+            int idx = defaultButtons.indexOf("COPYFROMSECTION");
+            defaultButtons.remove("COPYFROMSECTION");
+            defaultButtons.add(idx, "IPC_COPYFROMSECTION");
+        }
+
+        return defaultButtons;
+    }
```

Comment:
This method is repeated in many files. I recommend introducing a new superclass that handles this, and then extending that for each of the different sections.

Reply:
That makes sense. I need to improvise the code in many places, it was written in 19.1 version and project got shelved for different reasons. The new management is willing to resume the development slowly depending on how much functionality is already implemented. So, I am trying to restore the code into 21.7 and created a  new FB for this project. Thank you always for your feedback @labkey-jeckels.